peercrawler.py

```diff
@@ -82,6 +83,61 @@ def count_good_peers(self):
     def count_peers(self):
         return len(self.get_peers_copy())
 
+    def listen_incoming(self, port: int):
+        with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as s:
+            try:
+                s.bind(("::", port))
+                s.listen()
+
+                semaphore = threading.BoundedSemaphore(8)
+
+                while True:
+                    semaphore.acquire()
+                    connection, address = s.accept()
+                    threading.Thread(target=self.__handle_incoming_semaphore, args=(semaphore, connection, address), daemon=True).start()
+            except:
+                self.logger.exception("Error occurred in listener thread")
+                interrupt_main()
+
+    def __handle_incoming_semaphore(self, semaphore: threading.BoundedSemaphore, connection: socket.socket, address):
+        try:
+            self.handle_incoming(connection, address)
+        finally:
+            semaphore.release()
+
+    def handle_incoming(self, connection: socket.socket, address):
+        self.logger.debug(f"Receiving connection from {address}")
+
+        header, payload = get_next_hdr_payload(connection)
+        if header.msg_type == message_type(message_type_enum.node_id_handshake):
+            if header.is_response():
+                raise CommsError()
```

Comment:
We should print a log an every error handled.
We should also use the verbosity setting.
If an error happens frequently, it should only be printed when verbosity is 3 or above.